hips/tiles/tile.py

```diff
@@ -95,7 +95,7 @@ def read(cls, meta: HipsTileMeta, filename: str = None) -> 'HipsTile':
 
         if meta.file_format == 'fits':
             hdu_list = fits.open(str(path))
-            data = hdu_list[0].data
+            data = hdu_list[0].data.astype('float')
```

Comment:
What was the dtype before the conversion? Print out data.dtype.
Probably changing the data dtype on I/O isn't a good solution (because you want to be able to read and write back tiles unchanged).
So if needed at all this should happen when processing the tile for drawing, no?

In any case, please leave a comment why you're doing the `astype` call on the line above.